File: vida-segura-formio-python/test_forms.py
"""
Test Script - Verificar que la nueva arquitectura funciona
Prueba simple sin autenticación de usuario
"""

import sys
sys.path.insert(0, '.')

from src.application.form_service import form_service
from src.domain.form_schemas import get_form_schema
from src.shared.logger import logger


def test_forms():
    """Función principal de prueba"""
    try:
        logger.section('Testing Form Service')
        
        # Obtener el schema del formulario cliente
        cliente_schema = get_form_schema('cliente')
        
        logger.success('Schema loaded:', {
            'title': cliente_schema.get('title'),
            'name': cliente_schema.get('name'),
            'components': len(cliente_schema.get('components', []))
        })
        
        # Intentar listar formularios existentes
        logger.info('Listing existing forms...')
        forms = form_service.list_forms()
        
        logger.success('Forms retrieved:', {
            'count': len(forms) if isinstance(forms, list) else 0,
            'forms': [
                {'id': f.get('_id'), 'title': f.get('title')} 
                for f in forms[:3]
            ] if isinstance(forms, list) else []
        })
        
        logger.section('✅ All tests passed!')
    
    except Exception as error:
        logger.error('Test failed', error)
        sys.exit(1)


if __name__ == '__main__':
    test_forms()

File: test-forms.js
/**
 * Test Script - Verificar que la nueva arquitectura funciona
 * Prueba simple sin autenticación de usuario
 */

import { formService } from './vidaSegura/src/application/form-service.js';
import { getFormSchema } from './vidaSegura/src/domain/form-schemas.js';
import { logger } from './vidaSegura/src/shared/logger.js';

async function testForms() {
    try {
        logger.section('Testing Form Service');

        // Obtener el schema del formulario cliente
        const clienteSchema = getFormSchema('cliente');
        
        logger.success('Schema loaded:', {
            title: clienteSchema.title,
            name: clienteSchema.name,
            components: clienteSchema.components?.length || 0
        });

        // Intentar listar formularios existentes
        logger.info('Listing existing forms...');
        const forms = await formService.listForms();
        
        logger.success('Forms retrieved:', {
            count: forms?.length || 0,
            forms: forms?.slice(0, 3)?.map(f => ({ id: f._id, title: f.title }))
        });

        logger.section('✅ All tests passed!');
    } catch (error) {
        logger.error('Test failed', error);
        process.exit(1);
    }
}

testForms();
